### ISSUE TITLE
Tag Colors needing "Show Default Colors" is confusing

### ISSUE THREAD

**Issue by ferdnyc** [BiglySoftware/BiglyBT#514]
I think just since 1.5.0.1_B16 installed last night, I noticed that one of my tags was missing from the Tag Colors column. 

I finally worked out that switching on "Show Default Colors" brought it back — so I guess I hadn't customized the color of that tag. Once I discovered this, I went in to Tags Overview, selected the tag in question, and customized its color to the exact same color it had before... and now it's visible even without "Show Default Colors". OK, so... that happened.

It's kind of confusing to suddenly have that switch on, and tags you've been seeing all along suddenly disappear. What's more, it's kind of a hidden thing to have default to disabled, since there's very little to indicate to the user which tag(s) have default colors and won't be showing up in the column for that reason.

Also, it's entirely unclear how to reset a tag _back_ to the default color, if the user decides they **want** to exclude a tag from the list. I gues... is it the green swatch in the top row of the color picker? Is that the default color? There's absolutely nothing in the interface that would indicate that to the user other than trial-and-error. (AKA random flailing around, which is how I sorta worked that out... if indeed I _did_ work it out correctly.)

_Also_-also (sorry!), whenever that dialog comes up, there's always a checkmark in the red swatch at the _end_ of the top row. What's up with that?

Like, see how it came up when I clicked the Color box for my currently-pink tag: The top row is (I'm guessing) the current color, the default color, and then that red-with-a-checkmark color that _completely_ confuses me. What's its story?

![image](https://user-images.githubusercontent.com/538020/42380896-6e95cc1c-80fd-11e8-8d3a-0fc483414929.png)

#### System Information
```
Java 1.8.0_172 (64 bit)
  Oracle Corporation
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.172-9.b11.fc28.x86_64/jre

SWT v4757, gtk/3.22.30
Linux v4.17.2-200.fc28.x86_64, amd64 (64 bit)
B1.5.0.1_B16/4 az3
```

**Comment by parg** [comment]
You can reset a colour to default by right-clicking on the colour chooser button and selecting from the menu, As for that red checkmark - I have no idea - BiglyBT uses the SWT graphics library and this is supposed to use system widgets...

**Comment by ferdnyc** [comment]
> You can reset a colour to default by right-clicking on the colour chooser button and selecting from the menu,

Oh yeah! I never noticed that, thanks. 

From experimenting (with "Show Default Colors" turned off), it appears that although the green swatch in the upper strip of the color selector _contains_ the default color, selecting it isn't the same as SETTING the default color. It simply sets a custom color that's the same as the default color. The only way to _unset_ a tag's color is, as you say, "Reset Color" from the context menu of the Color button.

> As for that red checkmark - I have no idea - BiglyBT uses the SWT graphics library and this is supposed to use system widgets...

It's bizarre. Even more bizarre, I finally decided to just click "Select" in the color chooser on a tag, leaving the check mark in that red swatch and not selecting any other colors... it did set the tag's custom color to bright red, and the next time I opened the selector it looked like THIS:

![screenshot from 2018-07-12 05-13-01](https://user-images.githubusercontent.com/538020/42624120-9ecc7252-8592-11e8-9172-454c0729bb19.png)

That's the first time I've EVER seen that last swatch with the checkmark change from red to something else, and I _still_ have no idea what any of it means. In fact I'm more confused than ever! (Not that it really matters at all.) Anyway, thanks for the right-click tip!
